src/amplitude.js

```diff
@@ -362,53 +480,49 @@ Amplitude.prototype._getReferringDomain = function(referrer) {
   return null;
 };
 
-// since user properties are propagated on the server, only send once per session, don't need to send with every event
-Amplitude.prototype._saveReferrer = function(referrer) {
-  if (referrer === null || referrer === undefined || referrer === '') {
+/**
+ * Fetch the referrer information, parse the domain and send.
+ * Since user properties are propagated on the server, only send once per session, don't need to send with every event
+ * @private
+ */
+Amplitude.prototype._saveReferrer = function _saveReferrer(referrer) {
+  if (utils.isEmptyString(referrer)) {
     return;
   }
-
-  // always setOnce initial referrer
-  var referring_domain = this._getReferringDomain(referrer);
-  var identify = new Identify().setOnce('initial_referrer', referrer);
-  identify.setOnce('initial_referring_domain', referring_domain);
-
-  // only save referrer if not already in session storage or if storage disabled
-  var hasSessionStorage = false;
-  try {
-    if (window.sessionStorage) {
-      hasSessionStorage = true;
-    }
-  } catch (e) {
-    // utils.log(e); // sessionStorage disabled
-  }
-
-  if ((hasSessionStorage && !(this._getFromStorage(sessionStorage, LocalStorageKeys.REFERRER))) || !hasSessionStorage) {
-    identify.set('referrer', referrer).set('referring_domain', referring_domain);
-
-    if (hasSessionStorage) {
-      this._setInStorage(sessionStorage, LocalStorageKeys.REFERRER, referrer);
-    }
-  }
-
-  this.identify(identify);
+  var referrerInfo = {
+    'referrer': referrer,
+    'referring_domain': this._getReferringDomain(referrer)
+  };
+  _sendUserPropertiesOncePerSession(this, Constants.REFERRER, referrerInfo);
 };
 
-Amplitude.prototype.saveEvents = function() {
+/**
+ * Saves unsent events and identifies to localStorage. JSON stringifies event queues before saving.
+ * Note: this is called automatically every time events are logged, unless you explicitly set option saveEvents to false.
+ * @public
```

Comment:
I was initially a bit confused by what this meant.  Maybe an example use case would help here.